src/map/pet.c

```diff
@@ -345,7 +345,7 @@ static int pet_return_egg(struct map_session_data *sd, struct pet_data *pd)
 			pd->pet.pet_id == MakeDWord(sd->status.inventory[i].card[1], sd->status.inventory[i].card[2]));
 
 	if (i != MAX_INVENTORY) {
-		sd->status.inventory[i].identify = 1;
+		sd->status.inventory[i].attribute ^= ATTR_BROKEN;
```

Comment:
probably better unset attribute from broken state. to prevent situations if it was in wrong state